### PR TITLE
feat: add rake task to send emails to TOI candidates

### PR DESCRIPTION
**Pull request by shirleyodeng**
## Summary

This PR adds a rake task to send emails to candidates of TOI.

## Notion link

https://www.notion.so/talentprotocol/TOI-Custom-email-to-top-30-candidates-79f671b8ec0d4d9b945125f3baf1da15

## How to test?

run `rails notifications:send_toi_emails_to_top_candidates`

### CODE REVIEW

File: lib/tasks/notifications.rake [hunk 1]
```diff
@@ -17,4 +17,52 @@ namespace :notifications do
     SubscriptionMailer.with(recipient: user, source_id: user.id).subscription_accepted_email.deliver_later
     SponsorshipMailer.with(recipient: user, source_id: user.id).new_sponsor_email.deliver_later
   end
+
+  desc "Send TOI emails to top candidates"
+  task send_toi_emails_to_top_candidates: :environment do
+    organization = Organization.find_by!(slug: "takeoff-istanbul")
+    election = Election.find_by!(organization:)
+    sorted_candidates = Vote.where(election:).group_by(&:candidate_id).sort_by { |candidate_id, candidate_votes| candidate_votes.count }.reverse
```
Comment (shirleyodeng) [inline lib/tasks/notifications.rake:25]: Query shouldn't be too bad as there's 2598 votes in production.

File: lib/tasks/notifications.rake [hunk 2]
```diff
@@ -17,4 +17,52 @@ namespace :notifications do
...
+    top_ten = sorted_candidates.slice(0, 10)
+    tenth_vote_count = top_ten.last[1].count
+    eleventh_to_thirtieth = sorted_candidates.slice(10, 30)
+    eleventh_vote_count = eleventh_to_thirtieth.first[1].count
```
Comment (RubenSousaDinis) [inline lib/tasks/notifications.rake:29]: This is wrong. A `vote` record can have several votes. We use the `amount` field to store the number of votes a user got. Probably the issue is with the data design, but we can't fix it now 😄

Comment (shirleyodeng) [reply lib/tasks/notifications.rake]: @RubenSousaDinis fixed!

Comment (RubenSousaDinis) [reply lib/tasks/notifications.rake]: great!

**Review by RubenSousaDinis — APPROVED** [review]

**Pull request merged**